torch/csrc/Device.cpp

```diff
@@ -65,9 +65,6 @@ PyObject *THPDevice_pynew(PyTypeObject *type, PyObject *args, PyObject *kwargs)
     int32_t device_index = -1;
     if (!r.isNone(1)) {
       device_index = r.toInt64(1);
-      // -1 is allowed in ATen/C++, to mean the default device, but not in
-      // Python.
-      TORCH_CHECK(device_index >= 0, "Device index must not be negative");
     }
     at::Device device(as_device.type(), device_index);
     return THPDevice_New(device);
```

Comment:
I put a little effort into fixing the (existing) clang-tidy errors, but we might just have to ignore/suppress them. clang-tidy complains about global non-const objects, but it looks like PyBind expects these objects to be non-const (both the `PyTypeObject` constructor arguments and the call to `PyType_Ready()`). We could add the const-ness and cast it away, but "hiding" const-ness seems like a bad way to appease clang-tidy.